### DISCUSSION TITLE
Vercel open graph image generation with @vercel/og package only shows how to use next.js page router instead of the new app based router

### DISCUSSION THREAD

**Discussion by errorcodezero** [vercel/community#3255] [Bug Report]
### Page to Investigate

https://vercel.com/docs/frameworks/nextjs#open-graph-images

### Steps to Reproduce

If you go to the page and take a look at the example then you can see that it uses the pages directory. However, I would prefer to use the app router instead but the docs don't show how to do that.
![image](https://github.com/vercel/community/assets/74121237/a4bbc81f-184d-4a94-b72c-de08e70df749)

**Comment by hyjoong** [comment]
If you use the app router, I think it would be good to see `'import {ImageResponse} from 'next/server'`

https://nextjs.org/docs/app/api-reference/file-conventions/metadata/opengraph-image

**Comment by amyegan** [comment]
Thanks for pointing this out. For now, you can find examples using Next.js App Router and other frameworks on the [Open Graph Image Examples](https://vercel.com/docs/concepts/functions/edge-functions/og-image-generation/og-image-examples#) page. You can also find an example in the [Dynamic Image Generation](https://nextjs.org/docs/app/building-your-application/optimizing/metadata#dynamic-image-generation) and [opengraph-image](https://nextjs.org/docs/app/api-reference/file-conventions/metadata/opengraph-image) sections of the Next.js docs.
